hedera-node/hedera-network-admin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/handlers/NetworkGetAccountDetailsHandler.java

```diff
@@ -16,54 +16,271 @@
+import com.hedera.node.app.service.networkadmin.impl.config.NetworkAdminServiceConfig;
+import com.hedera.node.app.service.networkadmin.impl.utils.NetworkAdminServiceUtil;
+import com.hedera.node.app.service.token.ReadableAccountStore;
+import com.hedera.node.app.service.token.ReadableTokenRelationStore;
+import com.hedera.node.app.service.token.ReadableTokenStore;
+import com.hedera.node.app.service.token.ReadableTokenStore.TokenMetadata;
 import com.hedera.node.app.spi.workflows.PaidQueryHandler;
 import com.hedera.node.app.spi.workflows.PreCheckException;
 import com.hedera.node.app.spi.workflows.QueryContext;
+import com.hedera.node.config.data.LedgerConfig;
 import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
 import javax.inject.Inject;
 import javax.inject.Singleton;
 
 /**
- * This class contains all workflow-related functionality regarding {@link HederaFunctionality#GET_ACCOUNT_DETAILS}.
+ * This class contains all query information regarding {@link HederaFunctionality#GET_ACCOUNT_DETAILS}.
  */
 @Singleton
 public class NetworkGetAccountDetailsHandler extends PaidQueryHandler {
+
     @Inject
     public NetworkGetAccountDetailsHandler() {
-        // Exists for injection
+        // exists for injection
     }
 
     @Override
+    @NonNull
     public QueryHeader extractHeader(@NonNull final Query query) {
         requireNonNull(query);
-        return query.accountDetailsOrThrow().header();
+        return query.accountDetailsOrThrow().headerOrThrow();
     }
 
     @Override
+    @NonNull
     public Response createEmptyResponse(@NonNull final ResponseHeader header) {
         requireNonNull(header);
         final var response = GetAccountDetailsResponse.newBuilder().header(header);
         return Response.newBuilder().accountDetails(response).build();
     }
 
+    @Override
+    public boolean requiresNodePayment(@NonNull ResponseType responseType) {
+        return responseType == ANSWER_ONLY || responseType == ANSWER_STATE_PROOF;
+    }
+
+    @Override
+    public boolean needsAnswerOnlyCost(@NonNull ResponseType responseType) {
+        return COST_ANSWER == responseType;
+    }
+
     @Override
     public void validate(@NonNull final QueryContext context) throws PreCheckException {
         requireNonNull(context);
-        throw new UnsupportedOperationException("Not implemented");
+        final var query = context.query();
+        final var accountStore = context.createStore(ReadableAccountStore.class);
+        final GetAccountDetailsQuery op = query.accountDetailsOrThrow();
+        if (op.hasAccountId()) {
+            final var accountMetadata = accountStore.getAccountById(op.accountIdOrElse(AccountID.DEFAULT));
+            mustExist(accountMetadata, INVALID_ACCOUNT_ID);
+            if (accountMetadata.deleted()) {
+                throw new PreCheckException(INVALID_ACCOUNT_ID);
+            }
```

Comment:
If there is no accountId we need to return `INVALID_ACCOUNT_ID`

Reply:
as of right now we return   /**
   * There was a system error and the transaction failed because of invalid request parameters.
   */
  FAIL_INVALID = 23;

Reply:
I just as it in the mono code